downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,531 @@
+
+The object type Listener<T,A>, is described by the following object type descriptor:
+
+```ballerina
+object {
+   public function attach(T svc, A attachPoint) returns error?;
+   public function detach(T svc) returns error?;
+   public function start() returns error?;
+   public function gracefulStop() returns error?;
+   public function immediateStop() returns error?;
+}
+```
+
+##### Transactional Services
+
+Ballerina transaction capabilities are extended to services. Now, you can define transactional resource functions and transactional remote functions. These functions will be participants of global distributed transactions. Infection and agreement protocols are implemented based on the Ballerina distributed transaction protocol. 
+
+By defining services as participants, all services work as a single unit of work. If any of the services fails, the whole transaction will be reverted and if all the services are successfully called, the transaction will be completed and committed successfully. 
+
+**Defining a Transactional Resource Function of a Service**
+
+```ballerina
+transactional resource function get message(http:Caller caller, http:Request req) {
+    http:Response res = new;
+    callSomeService(res);        
+    callAnotherService(res);
+    checkpanic caller->respond(res);
+}
+```
+
+**Defining  the Transactional Remote Functions of a Client Object**
+
+
+```ballerina
+transactional remote function callMyFirstService() returns @tainted any|error {
+    return self.httpClient->get("/echo/message");
+}
+
+transactional remote function callMySecondService() returns @tainted any|error {
+    return self.httpClient->get("/echo/message");
+}
+```
+
+**Calling the Service**
+
+```ballerina
+transaction {
+    var res1 = client->callMyFirstService();
+   var  res2 = client ->callMySecondService();
+   var x = commit;
+    if (x is error) {
+        // error code
+    } else {
+        // success code
+    }
+}
+```                                                             
+
+#### Standard Library
+
+##### HTTP Module Changes
+
+###### Service Declaration
+
+- Basepath field has been removed from the `ServiceConfig` annotation. Use the `absolute resource path` that begins with `/` as the basePath which is optional and defaults to `/` when not specified.
+- The service type can be added as `http:Service`, which is optional after the `service` keyword.
+
+**Old Syntax**
+
+```ballerina
+@http:ServiceConfig {
+    basePath: “hello”
+}
+service myService on new http:Listener(9090) {
+
+}
+```
+
+**New Syntax**
+
+```ballerina
+service http:Service /hello on new http:Listener(9090) {
+
+}
+```
+
+###### Resource Method Declaration
+
+- Use the `resource accessor name` to specify the HTTP method instead of the `methods` field of the `ResourceConfig` (e.g., `get`).
+- Use `default` as the accessor name when any HTTP methods need to be supported (e.g., the passthrough/proxy use case).
```

Comment:
How about
```suggestion
- Use the resource method name to specify the HTTP method to support instead of the `methods` field of the `ResourceConfig` annotation (e.g., `get`).
- Use `default` as the resource method name when the resource has to support all the standard HTTP methods  (e.g., the passthrough/proxy use case).
```

Reply:
Just noticed the discussion on allowing custom methods, so how about
```suggestion
- Use the resource method name to specify the HTTP method to support instead of the `methods` field of the `ResourceConfig` annotation (e.g., `get`).
- Use `default` as the resource method name when the resource has to support all methods including standard HTTP methods and custom methods (e.g., the passthrough/proxy use case).
```